File: osrs-api/src/main/java/io/autorune/osrs/api/devices/MouseWheelListener.java
package io.autorune.osrs.api.devices;

import io.autorune.osrs.api.Client;

public interface MouseWheelListener extends MouseWheel, java.awt.event.MouseWheelListener {
	int getMouseWheelRotation();

	void setMouseWheelRotation(int value);

	Client getClientInstance();

	boolean getIsInputBlocked();

	void setIsInputBlocked(boolean value);
}

File: updater/src/main/kotlin/io/autorune/updater/analyzer/classes/implementations/devices/MouseWheelListener.kt
package io.autorune.updater.analyzer.classes.implementations.devices

import io.autorune.updater.analyzer.classes.ClassAnalyzer
import io.autorune.updater.analyzer.classes.annotations.CorrectFieldCount
import io.autorune.updater.analyzer.classes.annotations.CorrectMethodCount
import org.objectweb.asm.tree.ClassNode

@CorrectFieldCount(1)
@CorrectMethodCount(0)
class MouseWheelListener : ClassAnalyzer() {

    override fun findClassNode(): ClassNode? {
        for (node in classPool) {
            if (node.interfaces.contains("java/awt/event/MouseWheelListener"))
                return node
        }
        return null
    }

    private fun getMouseWheelRotation() {
        addField("mouseWheelRotation", classNode.fields.first { !isMemberStatic(it.access) && it.desc == "I" })
    }

    override fun getFields() {
        getMouseWheelRotation()
    }

}